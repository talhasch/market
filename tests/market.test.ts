
import { contractPrincipalCV, principalCV, uintCV } from "@stacks/transactions";
import { describe, expect, it } from "vitest";

const accounts = simnet.getAccounts();
const address1 = accounts.get("wallet_1")!;
const address2 = accounts.get("wallet_2")!;

const whitelisToken = () => {
  return simnet.callPublicFn("market", "whitelist-ft", [
    contractPrincipalCV(simnet.deployer, "token1")
  ], simnet.deployer);
}

describe("tests", () => {

  const expectMatchSnapshot = (msg: string, a: any) => {
    expect(a).toMatchSnapshot(msg)
  }

  it("basic trade test", () => {
    whitelisToken();

    expectMatchSnapshot('1- balances before trade', simnet.getAssetsMap());

    expectMatchSnapshot('2- order created',
      simnet.callPublicFn("market", "create-order", [
        uintCV(125_000000),
        uintCV(10_000000),
        contractPrincipalCV(simnet.deployer, "token1")
      ], address1).result);

    expectMatchSnapshot('3- balances after order creation', simnet.getAssetsMap());

    expectMatchSnapshot('4- order filled',
      simnet.callPublicFn("market", "fill-order", [
        uintCV(0),
        contractPrincipalCV(simnet.deployer, "token1")
      ], address2).result)

    expectMatchSnapshot('5- balances after trade', simnet.getAssetsMap())
  });



  /*
  it("token not whitelisted", () => {
    const { result } = simnet.callPublicFn("market", "create-order", [
      uintCV(125_000000),
      uintCV(15_340000),
      contractPrincipalCV(simnet.deployer, "token1")
    ], address1);

    console.log(result)
    

    expect(simnet.blockHeight).toBeDefined();

  });*/

  // it("shows an example", () => {
  //   const { result } = simnet.callReadOnlyFn("counter", "get-counter", [], address1);
  //   expect(result).toBeUint(0);
  // });
});
