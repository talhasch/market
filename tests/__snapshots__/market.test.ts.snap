// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests > basic trade test > 1- balances before trade 1`] = `
Map {
  ".token1.TOKEN1" => Map {
    "ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5" => 1000000000000n,
  },
  "STX" => Map {
    "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM" => 100000000000000n,
    "ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5" => 100000000000000n,
    "ST2CY5V39NHDPWSXMW9QDT3HC3GD6Q6XX4CFRK9AG" => 100000000000000n,
    "ST2JHG361ZXG51QTKY2NQCVBPPRRE2KZB1HR05NNC" => 100000000000000n,
    "ST2NEB84ASENDXKYGJPQW86YXQCEFEX2ZQPG87ND" => 100000000000000n,
    "ST2REHHS5J3CERCRBEPMGH7921Q6PYKAADT7JP2VB" => 100000000000000n,
    "ST3AM1A56AK2C1XAFJ4115ZSV26EB49BVQ10MGCS0" => 100000000000000n,
    "ST3NBRSFKX28FQ2ZJ1MAKX58HKHSDGNV5N7R21XCP" => 100000000000000n,
    "ST3PF13W7Z0RRM42A8VZRVFQ75SV1K26RXEP8YGKJ" => 100000000000000n,
    "STNHKEPYEPJ8ET55ZZ0M5A34J0R3N5FM2CMMMAZ6" => 100000000000000n,
  },
}
`;

exports[`tests > basic trade test > 2- order created 1`] = `
{
  "type": 7,
  "value": {
    "type": 1,
    "value": 0n,
  },
}
`;

exports[`tests > basic trade test > 3- balances after order creation 1`] = `
Map {
  ".token1.TOKEN1" => Map {
    "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.market" => 125000000n,
    "ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5" => 999875000000n,
  },
  "STX" => Map {
    "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM" => 100000000000000n,
    "ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5" => 100000000000000n,
    "ST2CY5V39NHDPWSXMW9QDT3HC3GD6Q6XX4CFRK9AG" => 100000000000000n,
    "ST2JHG361ZXG51QTKY2NQCVBPPRRE2KZB1HR05NNC" => 100000000000000n,
    "ST2NEB84ASENDXKYGJPQW86YXQCEFEX2ZQPG87ND" => 100000000000000n,
    "ST2REHHS5J3CERCRBEPMGH7921Q6PYKAADT7JP2VB" => 100000000000000n,
    "ST3AM1A56AK2C1XAFJ4115ZSV26EB49BVQ10MGCS0" => 100000000000000n,
    "ST3NBRSFKX28FQ2ZJ1MAKX58HKHSDGNV5N7R21XCP" => 100000000000000n,
    "ST3PF13W7Z0RRM42A8VZRVFQ75SV1K26RXEP8YGKJ" => 100000000000000n,
    "STNHKEPYEPJ8ET55ZZ0M5A34J0R3N5FM2CMMMAZ6" => 100000000000000n,
  },
}
`;

exports[`tests > basic trade test > 4- order filled 1`] = `
{
  "type": 7,
  "value": {
    "type": 1,
    "value": 0n,
  },
}
`;

exports[`tests > basic trade test > 5- balances after trade 1`] = `
Map {
  ".token1.TOKEN1" => Map {
    "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.market" => 0n,
    "ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5" => 999875000000n,
    "ST2CY5V39NHDPWSXMW9QDT3HC3GD6Q6XX4CFRK9AG" => 125000000n,
  },
  "STX" => Map {
    "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM" => 100000000100000n,
    "ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5" => 100000009900000n,
    "ST2CY5V39NHDPWSXMW9QDT3HC3GD6Q6XX4CFRK9AG" => 99999990000000n,
    "ST2JHG361ZXG51QTKY2NQCVBPPRRE2KZB1HR05NNC" => 100000000000000n,
    "ST2NEB84ASENDXKYGJPQW86YXQCEFEX2ZQPG87ND" => 100000000000000n,
    "ST2REHHS5J3CERCRBEPMGH7921Q6PYKAADT7JP2VB" => 100000000000000n,
    "ST3AM1A56AK2C1XAFJ4115ZSV26EB49BVQ10MGCS0" => 100000000000000n,
    "ST3NBRSFKX28FQ2ZJ1MAKX58HKHSDGNV5N7R21XCP" => 100000000000000n,
    "ST3PF13W7Z0RRM42A8VZRVFQ75SV1K26RXEP8YGKJ" => 100000000000000n,
    "STNHKEPYEPJ8ET55ZZ0M5A34J0R3N5FM2CMMMAZ6" => 100000000000000n,
  },
}
`;
